src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends T> type) {
+        List<Machine> machines = new ArrayList<>();
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                return machines = new BulldozerProducer().get();
+            case "Excavator":
+                return machines = new ExcavatorProducer().get();
+            case "Truck":
+                return machines = new TruckProducer().get();
+            default:
+                return machines;
+        }
```

Comment:
I mean we don't  need line List of machines = new ArrayList<>():
We can write for example "return new BulldozerProducer().get()" so 10 line will be redundant.